### PR TITLE
#184347109 added payment with stripe

### PR DESCRIPTION
**Pull request by teerenzo**
### What does this PR do?

This PR intend to allow buyer to make payment 

### Description of Task to be completed?

User will be able to make payment after successfully payment, cart will be cleared and User will be able to see ordered products

### How should this be manually tested?

you can test manual by using provided end point on docs

### Any background context you want to provide?

Any other information you can give about the task.

### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/184347109

### Questions:

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code is formatted correctly
- [x] My changes doesn't have any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]
1. Your PR is named wrong
2. your commit is named wrong

File: src/controllers/paymentController.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable no-else-return, require-jsdoc */
+
+import PaymentService from "../services/paymentService";
+
+class PaymentController {
+  static async createCheckoutSession(req, res) {
+    await PaymentService.createCheckoutSession(req, res);
+  }
+
+  static async success(req, res) {
```
Comment (Eli250) [inline src/controllers/paymentController.js]: The naming of these functions is not clear. You have to follow conventions.

File: src/routes/api/payment.route.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import express from "express";
+import extractToken from "../../middlewares/extractToken";
+import checkRole from "../../middlewares/checkRole";
+import PaymentController from "../../controllers/paymentController";
+import IsUserHasCart from "../../middlewares/checkUserCartExist";
+
+const route = express.Router();
+
+route.post("/",extractToken,checkRole(["buyer"]),IsUserHasCart, PaymentController.createCheckoutSession);
+route.get("/success",PaymentController.success)
```
Comment (Eli250) [inline src/routes/api/payment.route.js]: These routes have no meaning. Use something everyone can understand

File: src/services/paymentService.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+class PaymentService {
+  static async createCheckoutSession(req, res) {
+    const appUrl = `${req.protocol}://${req.headers.host}`;
+    console.log(appUrl);
+    const token = JwtUtil.generate(req.user);
+
+    try {
+      const cart = await Cart.findOne({ where: { buyerId: req.user.id } });
+      const lineItems = [];
```
Comment (Eli250) [inline src/services/paymentService.js]: What is lineItems? Use a self-explanatory variable.

File: src/services/paymentService.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+const stripe = require("stripe")(
+  `${process.env.STRIPE_KEY}`
+);
+
+const { Cart, Order, Product, Sales } = require("../database/models");
+
+class PaymentService {
+  static async createCheckoutSession(req, res) {
+    const appUrl = `${req.protocol}://${req.headers.host}`;
+    console.log(appUrl);
```
Comment (Eli250) [inline src/services/paymentService.js]: Remove this console.

**Review by Eli250 — CHANGES_REQUESTED** [review]

File: src/routes/api/product.route.js [hunk 5]
```diff
@@ -33,18 +32,16 @@ route.get(
   ProductController.getAllItems
 );
 route.get(
-  "/:id",
+  "/all/productWishes",
```
Comment (Eli250) [inline src/routes/api/product.route.js]: This endpoint does not make sense

**Review by Eli250 — APPROVED** [review]

**Review by Ntarecedrick — APPROVED** [review]

**Review by NG-Solennel — APPROVED** [review]

**Pull request merged**